Sources/Swift/SentryCrash/SentryCrashWrapper.swift

```diff
@@ -0,0 +1,291 @@
+        deviceData["usable_memory"] = systemInfo["usableMemorySize"]
+        deviceData["memory_size"] = systemInfo["memorySize"]
+        
+        deviceData["locale"] = Locale.autoupdatingCurrent.identifier
+        
+        // Set screen dimensions if available
+        setScreenDimensions(&deviceData)
+        
+        scope.setContext(value: deviceData, key: "device")
+    }
+    
+    private func enrichScopeWithAppData(_ scope: Scope, systemInfo: [String: Any]) {
+        var appData: [String: Any] = [:]
+        let infoDict = Bundle.main.infoDictionary ?? [:]
+        
+        appData["app_identifier"] = infoDict["CFBundleIdentifier"]
+        appData["app_name"] = infoDict["CFBundleName"]
+        appData["app_build"] = infoDict["CFBundleVersion"]
+        appData["app_version"] = infoDict["CFBundleShortVersionString"]
+        
+        appData["app_start_time"] = systemInfo["appStartTime"]
+        appData["device_app_hash"] = systemInfo["deviceAppHash"]
+        appData["app_id"] = systemInfo["appID"]
+        appData["build_type"] = systemInfo["buildType"]
+        
+        scope.setContext(value: appData, key: "app")
+    }
+    
+    private func enrichScopeWithRuntimeData(_ scope: Scope) {
+        var runtimeContext: [String: Any] = [:]
+        
+        // We set this info on the runtime context because the app context has no existing fields
+        // suitable for representing Catalyst or iOS-on-Mac execution modes. We also wanted to avoid
+        // adding two new Apple-specific fields to the app context. Coming up with a generic,
+        // reusable property on the app context proved difficult, so instead we reuse the "name"
+        // field of the runtime context as a pragmatic and semantically acceptable solution.
+        // isiOSAppOnMac and isMacCatalystApp are mutually exclusive, so we only set one of them.
+        if #available(iOS 14.0, macOS 11.0, watchOS 7.0, tvOS 14.0, *) {
+            if self.processInfoWrapper.isiOSAppOnMac {
+                runtimeContext["name"] = "iOS App on Mac"
+                runtimeContext["raw_description"] = "ios-app-on-mac"
+            }
+        }
+        
+        if #available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *) {
+            if self.processInfoWrapper.isMacCatalystApp {
+                runtimeContext["name"] = "Mac Catalyst App"
+                runtimeContext["raw_description"] = "raw_description"
+            }
+        }
+        
+        if !runtimeContext.isEmpty {
+            scope.setContext(value: runtimeContext, key: "runtime")
+        }
+    }
+    
+    private func getOSName() -> String {
+#if os(macOS) || targetEnvironment(macCatalyst)
+        return "macOS"
+#elseif os(iOS)
+        return "iOS"
+#elseif os(tvOS)
+        return "tvOS"
+#elseif os(watchOS)
+        return "watchOS"
+#elseif (swift(>=5.9) && os(visionOS))
+        return "visionOS"
+#else
+        return "Unknown"
```

Comment:
`m`: Unknown is a new functionality. Previously, we didn't set the OS if we couldn't detect it. Ist that by intention?